src/main/java/com/github/tombentley/klog/segment/cli/TxnStat.java

```diff
@@ -123,6 +125,14 @@ private void printOpen(boolean filenames, ProducerSession sess, FirstBatchInTxn
             System.out.printf("%d:", txn.firstBatchInTxn().line());
         }
         System.out.printf(" %s->%s%n", sess, txn);
+        if (emptyTransaction != null) {
+            System.out.printf("Found a matching out of order control batch: %s\n", emptyTransaction);
+            System.out.print("To forcibly abort this transaction, run: ");
+            System.out.print("bin/kafka-transactions.sh --bootstrap-server <BOOTSTRAP_SERVER> --command-config <CONFIG_FILE> abort ");
+            System.out.print("--topic <TOPIC> --partition <PARTITION> ");
+            System.out.printf("--producer-id %d --producer-epoch %d ", sess.producerId(), sess.producerId());
```

Comment:
There is a trivial issue here. you are passing twice the sess.producerId() instead of the epoch

Reply:
Good catch! Thanks @saiello. After a bit more testing it turns out there are a few issues with this PR so I'm closing it.